lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -822,26 +822,37 @@ circt::firrtl::maybeStringToLocation(StringRef spelling, bool skipParsing,
 /// Given a type, return the corresponding lowered type for the HW dialect.
 /// Non-FIRRTL types are simply passed through. This returns a null type if it
 /// cannot be lowered.
-Type circt::firrtl::lowerType(Type type) {
-  auto firType = firrtl::type_dyn_cast<FIRRTLBaseType>(type);
+Type circt::firrtl::lowerType(
+    Type type, std::optional<Location> loc,
+    const std::function<hw::TypeAliasType(Type, BaseTypeAliasType, Location)>
+        &getTypeDeclFn) {
+  auto firType = type.dyn_cast<FIRRTLBaseType>();
   if (!firType)
     return type;
 
+  if (BaseTypeAliasType aliasType =
+          firrtl::type_dyn_cast<BaseTypeAliasType>(firType)) {
```

Comment:
type_dyn_cast strips outermost type aliases. To access type alias, it's necessary to use normal dyn_cast.